htdocs/api/v0.0.2/APIBase.php

```diff
@@ -143,9 +143,9 @@ function handleETag()
     /**
      * Calculate the ETag for this resource
      *
-     * @return string an ETag for this resource
+     * @return ?string an ETag for this resource
      */
-    abstract function calculateETag();
+    abstract function calculateETag(): ?string;
```

Comment:
I think this should always return a string (the empty string if there's no etag, rather than null), but as @xlecours says this is going to be dead code once the API PR gets merged and the api directory is ignored by phan, so we don't need to do anything about this.